app/controllers/charges_controller.rb

```diff
@@ -9,25 +9,36 @@ def new
   end
 
   def create
+    # I know this controller action is fat. It will be refactored in a future PR
     order = current_user.orders.pending.last
     if order
       # Amount in cents
       amount = order.stripe_total
-      customer = Stripe::Customer.create(
-        email:  current_user.email,
-        source: params[:stripeToken]
-      )
+
+      if session[:stripeToken]
+        customer = Stripe::Customer.create(
```

Comment:
Could you explain how the credit card saving feature works? I thought it would be attach to a customer in stripe, but I haven't implemented stripe in some time.